src/bson-encode.c

```diff
@@ -648,6 +648,44 @@ void php_phongo_zval_to_bson(zval* data, php_phongo_bson_flags_t flags, bson_t*
 	php_phongo_field_path_free(field_path);
 } /* }}} */
 
+/* Converts the array or object argument to a bson_value_t. If the object is an
```

Comment:
This doesn't seem to care about the type of `zval *data`. Should it be checking that and throw if it's not `IS_OBJECT` or `IS_ARRAY`? If not, should this be rephrased?

Reply:
Whoops. Rephrased it as the function is capable of handling any arbitrary type which can be converted to a `bson_value_t`.